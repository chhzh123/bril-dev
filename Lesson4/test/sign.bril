# ARGS: sign

@main {
  a: int = const 10;
  b: int = const -5;
  cond: bool = const true;
  br cond .left .right;
.left:
  c: int = add a b;
  d: int = mul a b;
  jmp .end;
.right:
  one: int = const 1;
  c: int = add a one;
  jmp .end;
.end:
  print c;
}
