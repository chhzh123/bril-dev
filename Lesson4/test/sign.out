b1:
  in:  {'a': '⊥', 'b': '⊥', 'cond': '⊥', 'c': '⊥', 'd': '⊥', 'one': '⊥'}
  out: {'a': '+', 'b': '-', 'cond': '+', 'c': '⊥', 'd': '⊥', 'one': '⊥'}
left:
  in:  {'a': '+', 'b': '-', 'cond': '+', 'c': '⊥', 'd': '⊥', 'one': '⊥'}
  out: {'a': '+', 'b': '-', 'cond': '+', 'c': 'T', 'd': '-', 'one': '⊥'}
right:
  in:  {'a': '+', 'b': '-', 'cond': '+', 'c': '⊥', 'd': '⊥', 'one': '⊥'}
  out: {'a': '+', 'b': '-', 'cond': '+', 'c': '+', 'd': '⊥', 'one': '+'}
end:
  in:  {'a': '+', 'b': '-', 'cond': '+', 'c': 'T', 'd': '⊥', 'one': '⊥'}
  out: {'a': '+', 'b': '-', 'cond': '+', 'c': 'T', 'd': '⊥', 'one': '⊥'}